pkg/controller/storagecluster/reconcile.go

```diff
@@ -1152,3 +1156,134 @@ func getMonCount(nodeCount int) int {
 
 	return count
 }
+
+// ensureJobTemplates ensures if the osd removal job template exists
+func (r *ReconcileStorageCluster) ensureJobTemplates(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	osdCleanUpTemplate := &openshiftv1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name + "-osd-removal",
+			Namespace: sc.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, osdCleanUpTemplate, func() error {
+		osdCleanUpTemplate.Objects = []runtime.RawExtension{
+			{
+				Object: newCleanupJob(sc),
+			},
+		}
+		osdCleanUpTemplate.Parameters = []openshiftv1.Parameter{
+			{
+				Name:     "ROOK_IMAGE",
+				Required: true,
+				Value:    os.Getenv("ROOK_CEPH_IMAGE"),
+			},
+			{
+				Name:     "FAILED_OSD_ID",
+				Required: true,
+			},
+		}
+		return controllerutil.SetControllerReference(sc, osdCleanUpTemplate, r.scheme)
+	})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create Template: %v", err.Error())
+	}
+	return nil
+}
+
+func newCleanupJob(sc *ocsv1.StorageCluster) *batchv1.Job {
+	labels := map[string]string{
+		"app": "ceph-toolbox-job-${FAILED_OSD_ID}",
+	}
+
+	job := &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "batch/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-toolbox-job-${FAILED_OSD_ID}",
+			Namespace: sc.Namespace,
+			Labels:    labels,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "mon-endpoint-volume",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "mon-endpoint-volume",
+									},
+									Items: []corev1.KeyToPath{
+										{
+											Key:  "data",
+											Path: "mon-endpoint",
+										},
+									},
+								},
+							},
+						},
+						{
+							Name:         "ceph-config",
+							VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:  "script",
+							Image: "${ROOK_IMAGE}",
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									MountPath: "/etc/ceph",
+									Name:      "ceph-config",
+									ReadOnly:  true,
+								},
+							},
+							Command: []string{"/bin/bash", "-c",
+								`# ensure the osd is marked out
+								 ceph osd out osd.FAILED_OSD_ID
+								 # purge the osd 
```

Comment:
Note that the script will have lots of spaces before these lines. It shouldn't hurt the script, but it would be better if we could avoid that whitespace.

Reply:
ack